aiida/orm/nodes/process/calculation/calcjob.py

```diff
@@ -163,11 +163,16 @@ def _validate(self):
             raise exceptions.ValidationError('invalid calculation state `{}`'.format(self.get_state()))
 
         try:
-            self.get_parser_class()
-        except exceptions.MissingPluginError:
-            raise exceptions.ValidationError("No valid class/implementation found for the parser '{}'. "
-                                             "Set the parser to None if you do not need an automatic "
-                                             "parser.".format(self.get_option('parser_name')))
+            parser_class = self.get_parser_class()
+        except (exceptions.MissingEntryPointError, exceptions.MultipleEntryPointError,
+                exceptions.LoadingEntryPointError) as exception:
+            raise exceptions.ValidationError('invalid parser specified: {}'.format(exception))
+
+        try:
+            if parser_class is not None:
+                parser_class(self)
```

Comment:
`None(self)` also raises a `TypeError`.
If there is a reason why the `parser_class` can be `None`, could you put a comment here explaining what this means?

Reply:
It can be `None` because we don't require a parser to be defined. I will put a comment